From e677865f6ccb0c719af423999402319991f7272f Mon Sep 17 00:00:00 2001
From: Ezra Buehler <ezra.buehler@husqvarnagroup.com>
Date: Fri, 14 Feb 2025 00:13:57 +0100
Subject: [PATCH] net: ethernet: ralink: Replace ethtool_puts with memcpy

Revert to the initial implementation, ethtool_puts() does not exist
upstream.

Upstream-Status: Inappropriate
---
 drivers/net/ethernet/ralink/ethtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ralink/ethtool.c b/drivers/net/ethernet/ralink/ethtool.c
index 5d4cebb08968..599a0d36e14b 100644
--- a/drivers/net/ethernet/ralink/ethtool.c
+++ b/drivers/net/ethernet/ralink/ethtool.c
@@ -165,7 +165,7 @@ static void fe_get_strings(struct net_device *dev, u32 stringset, u8 *data)
 {
 	switch (stringset) {
 	case ETH_SS_STATS:
-		ethtool_puts(&data, *fe_gdma_str);
+		memcpy(data, *fe_gdma_str, sizeof(fe_gdma_str));
 		break;
 	}
 }
-- 
2.43.0

